cfpb/cfgov-refresh
# Make Resources searchable and filterable, convert Snippet Lists to Resource Lists

This started as simply wanting to make Resource snippets searchable and filterable (/cc @sonnakim) and evolved into simplifying Snippet Lists and renaming them to Resource Lists, as well. Review the first commit to see the work done for setting up a ModelAdmin view for Resources, and the subsequent commits to see the disentangling of Resources from Snippets.

Refer to GHE issues https://GHE/CFGOV/platform#2984 and https://GHE/CFPB/Hubcap#62 for more background.

Many thanks to @chosak for pairing with me on various pieces of this.

## Additions

- ModelAdmin view of Resources that supports searching and filtering them

## Changes

- Resources are no longer registered as snippets
- Snippet Lists have been renamed to Resource Lists and no longer pretend to support any other types of snippets (which they never really did)
- Fixes Snippet List header cell background color

## Testing

(assumes recent DB loaded)

1. Pull branch
1. `gulp styles`
1. `./cfgov/manage.py migrate`
1. Load an existing page that has a Resource List (née Snippet List) and confirm it still loads as expected, e.g., [MSEM](http://localhost:8000/consumer-tools/managing-someone-elses-money/) or [library outreach materials](http://localhost:8000/practitioner-resources/library-resources/outreach-materials-to-use-outside-the-library/)
1. Edit one of the Resource Lists on one of those pages and confirm that things save correctly and still load
1. Go to the new **Resources** item in the main Wagtail sidebar
1. Play around with searching & filtering
1. Use search or filters to edit one of the resources on one of the lists you saw before
1. Reload the page and see that the edits correctly took effect
1. Run the tests with `tox -e fast`

## Screenshots

![screen shot 2018-09-19 at 12 53 34](https://user-images.githubusercontent.com/1044670/45768492-0f2bf000-bc0b-11e8-8a7f-e7a4db755871.png)

## Notes

- To avoid the necessity for a data migration, I left the name of the StreamField block in BrowsePage and SublandingPage as `snippet_list`. If people feel strongly about it, I could rename and do the migration dance, but I know people are anxious to get these features available.

## Todos

- [ ] Notify Wagtail editors about this change via docs/announcement (/cc @schaferjh)

## Checklist
